app/forms/declaration/WarehouseIdentification.scala

```diff
@@ -57,8 +57,16 @@ object WarehouseIdentification {
           )
       )
     )(WarehouseIdentification.apply)(WarehouseIdentification.unapply)
+    .verifying("supplementary.warehouse.identificationType.error", typeSelectedWhenNumberIsPopulated)
     .verifying("supplementary.warehouse.identificationNumber.empty", idNumberIsPopulatedWhenIDTypeIsSelected)
 
+  private def typeSelectedWhenNumberIsPopulated: WarehouseIdentification => Boolean =
+    w =>
+      w.identificationNumber match {
```

Comment:
w.identificationType.exists(_.nonEmpty) || w.identificationType.isEmpty

Reply:
I think you mean 
`w => w.identificationNumber.isEmpty || w.identificationType.exists(_.nonEmpty)`
I just "matched" the style of the existing `idNumberIsPopulatedWhenIDTypeIsSelected` method but I can change both if you prefer.

Reply:
Whichever you think reads best